client/js/socket-events/changelog.js

```diff
@@ -4,19 +4,70 @@ const $ = require("jquery");
 const socket = require("../socket");
 const templates = require("../../views");
 
+module.exports = {
+	requestIfNeeded,
+};
+
+// Requests version information if it hasn't been retrieved before (or if it has
+// been removed from the page, i.e. when clicking on "Check now". Displays a
+// loading state until received.
+function requestIfNeeded() {
+	if ($("#version-checker").is(":empty")) {
+		renderVersionChecker({status: "loading"});
+		socket.emit("changelog");
+	}
+}
+
 socket.on("changelog", function(data) {
-	const container = $("#changelog-version-container");
+	// 1. Release notes window for the current version
+	$("#changelog").html(templates.windows.changelog(data.current));
+
+	const links = $("#changelog .changelog-text a");
+	// Make sure all links will open a new tab instead of exiting the application
+	links.attr("target", "_blank");
+	// Add required metadata to image links, to support built-in image viewer
+	links.has("img").addClass("toggle-thumbnail");
+
+	// 2. Version checker visible in Help window
+	let status;
 
 	if (data.latest) {
-		container.addClass("new-version");
-		container.html(templates.new_version(data));
+		status = "new-version";
 	} else if (data.current.changelog) {
-		container.addClass("up-to-date");
-		container.text("The Lounge is up to date!");
+		status = "up-to-date";
 	} else {
-		container.addClass("error");
-		container.text("An error has occurred, try to reload the page.");
+		status = "error";
 	}
 
-	$("#changelog").html(templates.windows.changelog(data));
+	renderVersionChecker({
+		latest: data.latest,
+		status,
+	});
+
+	// When there is a button to refresh the checker available, display it when
+	// data is expired. Before that, server would return same information anyway.
+	if (data.expiresAt) {
+		setTimeout(
+			() => $("#version-checker").find("#check-now").show(),
```

Comment:
This can be a single selector.

Reply:
Good catch, done! Also checked that this was still working.